plugins/org.eclipse.n4js/src/org/eclipse/n4js/postprocessing/PolyProcessor.xtend

```diff
@@ -175,6 +199,73 @@ package class PolyProcessor extends AbstractPolyProcessor {
 		infCtx.solve;
 	}
 
+	/** Calculate expected type of a destructure pattern based on its structure */
+	private def TypeRef calculateExpectedType(DestructNode destructNode, RuleEnvironment G) {
```

Comment:
move this and next method to DestructureHelper?

Reply:
done.